--- rosegarden-1/data/desktop/com.rosegardenmusic.rosegarden.desktop	2024-06-07 18:10:23.940496943 +0100
+++ rosegarden-1/data/desktop/com.rosegardenmusic.rosegarden.desktop	2024-06-07 18:10:48.020988772 +0100
@@ -4,7 +4,7 @@
 Exec=rosegarden %f
 StartupWMClass=Rosegarden
 MimeType=audio/x-rosegarden-composition;audio/x-rosegarden-device;audio/x-rosegarden-project;audio/x-rosegarden-template;audio/midi;
-Keywords=midi audio sequencer music notation score
+Keywords=midi;audio;sequencer;music;notation;score;
 X-KDE-NativeMimeType=audio/x-rosegarden-composition
 Icon=rosegarden
 Comment=MIDI and Audio Sequencer and Notation Editor
